### DISCUSSION TITLE
Why Symfony needs a frontend framework.

### DISCUSSION THREAD

**Discussion by excitedbox** [symfony/symfony#43336] [Q&A]
Going beyond the fact that most JS framework options are meant to integrate with other JS frameworks or offer their own solutions to routing etc. Symfony uses several non standard components as far as JS is concerned such as Doctrine, Twig, Stimulus and Symfony itself. This means either hacking together our own solution or building from scratch. 

If using Symfony comes with the added complexity of integrating another JS framework it is easier, less error prone, and faster to use the options already provided by JS Frameworks. Building an app is work enough, having to modify JS components to work with doctrine and rebuilding components using twig seems a waste of time and additional work. This could lead to twig dying, only being used to fill variables or only being used for displaying content that doesn't fit default components.

My Proposal:
I propose a set of reusable widgets/components built using Twig, Stimulus, Doctrine, and Tailwind turning Symfony into a full stack solution for modern, fast, app development. Symfony UX already provides charts and a few other items. Things missing are data grids, image/file libraries, diagrams, and many of the standard/reused components needed for displaying and editing data.

If we orient ourselves on the widget lists of frameworks like Webix, Vaadin or Kendo we would have a great set of components that can be quickly integrated and styled to fit our needs.

Not only would this help Symfony gain many new users, most of the "complete" component frameworks only include the data components in paid tiers. This could either help draw more users or add a revenue source for more Symfony development.

**Comment by herndlm** [comment]
As this is no issue it sounds like something that should/could be a discussion at https://github.com/symfony/symfony/discussions?

**Comment by excitedbox** [comment]
After spending the last few days looking into a solution to this issue I have come across the Bolt Design System which seems like a great fit for integration into Symfony. It is Based on Twig, web components and a modified Pattern Labs workflow using yaml to set design configurations.

The website:
https://boltdesignsystem.com/
A talk demonstrating the benefits:
https://www.youtube.com/watch?v=NhjEbuuKn1Y

Integrating/modifying this for Symfony would be a huge benefit for the community. Everywhere I searched symfony/twig ui components, the current lack of frontend components was mentioned as the main reason for not choosing symfony. I believe this would bring a lot of new life into the project.
